cmake/GenerateVersion.cmake

```diff
@@ -1,11 +1,13 @@
 set(NVIM_VERSION
     "v${NVIM_VERSION_MAJOR}.${NVIM_VERSION_MINOR}.${NVIM_VERSION_PATCH}${NVIM_VERSION_PRERELEASE}")
 
+if(EXISTS "${NVIM_SOURCE_DIR}/.git")
 execute_process(
   COMMAND git --git-dir=${NVIM_SOURCE_DIR}/.git --work-tree=${NVIM_SOURCE_DIR} describe --first-parent --dirty --always
   OUTPUT_VARIABLE GIT_TAG
   OUTPUT_STRIP_TRAILING_WHITESPACE
```

Comment:
I'd prefer just slapping on a `ERROR_QUIET` to `execute_process` instead.

Reply:
Would it be better to silent an error rather than fixing it?

Reply:
Your PR gives `v0.9.0-dev-` without a `.git` directory.

Reply:
I think so, yes. I don't think it's worth having accounting for every possible scenario.

Also, your PR gives `v0.9.0-dev-` without a `.git` repo, it should be `v0.9.0-dev`.

Reply:
I have added the change as you suggested.

> your PR gives v0.9.0-dev- without a .git repo

It could be fixed by adding the same check with the `if (NVIM_VERSION_PRERELEASE)` condition. Or checking `GIT_TAG` after execute_process.